### PR TITLE
Add dnd? method for User

### PR DESCRIPTION
**Pull request by kaine119**
# Summary
Adds a `User#dnd?` method alongside the rest of the `status?` methods. Also adds documentation for each method

---
## Added
* `User#dnd?`
* Documentation for `User#online?`, `offline?`, `idle?` and `dnd?`

## Changed
* Add grouping around the methods above (feel free to revert this before squashing)

### CODE REVIEW

File: lib/discordrb/data/user.rb [hunk 1]
```diff
@@ -166,7 +166,18 @@ def webhook?
       @discriminator == Message::ZERO_DISCRIM
     end
 
-    %i[offline idle online].each do |e|
+    # @!group Statuses
```
Comment (swarley) [inline lib/discordrb/data/user.rb]: I'm not a fan of the group here personally, but if we keep it the syntax needs a fixup
```suggestion
    # @!group Statuses

```

File: lib/discordrb/data/user.rb [hunk 2]
```diff
@@ -166,7 +166,18 @@ def webhook?
...
+    # @!method offline?
+    #   @return [true, false] whether this user is offline.
+    # @!method idle?
+    #   @return [true, false] whether this user is idle.
+    # @!method online?
+    #   @return [true, false] whether this user is online.
+    # @!method dnd?
+    #   @return [true, false] whether this user is on Do Not Disturb.
+    # @!endgroup
```
Comment (swarley) [inline lib/discordrb/data/user.rb]:
```suggestion

    # @!endgroup
```

If we do keep the group directive, the beginning and end must be in separate declarations for YARD to render it properly

File: lib/discordrb/data/user.rb [hunk 3]
```diff
@@ -166,7 +166,18 @@ def webhook?
...
-    %i[offline idle online].each do |e|
+    # @!group Statuses
+    # @!method offline?
+    #   @return [true, false] whether this user is offline.
+    # @!method idle?
+    #   @return [true, false] whether this user is idle.
+    # @!method online?
+    #   @return [true, false] whether this user is online.
+    # @!method dnd?
+    #   @return [true, false] whether this user is on Do Not Disturb.
```
Comment (swarley) [inline lib/discordrb/data/user.rb]: not sure how I feel about the pascal casing for `do not disturb`. maybe something like this?
```suggestion
    #   @return [true, false] whether this user is set to do not disturb.
```
I'm sure this is opinionated though

**Comment by z64** [comment]
Thank you!

**Pull request merged**